services/request_service.go

```diff
@@ -169,6 +166,45 @@ func (rs RequestService) Dereference(didUrl string, dereferenceOptions types.Der
 	return didDereferencing
 }
 
+func (rs RequestService) dereferenceService(did string, query string, fragmentId string, didUrl string, dereferenceOptions types.DereferencingOption) (types.DidDereferencing, error) {
```

Comment:
`didUrl` already contains `did string, query string, fragmentId string`. Can we use it?